pkg/azure/client_test.go

```diff
@@ -0,0 +1,46 @@
+// +build unittest
+
+package azure
+
+import (
+	"errors"
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+	"k8s.io/klog/v2"
+	"net/http"
+	"time"
+)
+
+type FakeSender struct {
+	statusCode int
+}
+
+func (fs FakeSender) Do(request *http.Request) (response *http.Response, err error) {
+	response = &http.Response{
+		StatusCode: fs.statusCode,
+	}
+	if fs.statusCode != 200 {
+		err = errors.New("Error while making a GET for the gateway")
+	}
+	return response, err
+}
+
+var _ = DescribeTable("Az Application Gateway failures using authorizer", func(statusCodeArg int) {
```

Comment:
How about
`func(statusCodeArg int, errorExpected bool)`

if errorExpected {
  Expect(err).To(HaveOccurred())
} else {
  Expect(err).ToNot(HaveOccurred())
}

Reply:
I really like this suggestion! Thanks! :) I've implemented it.